### PR TITLE
Skip Audit - Painting Library

### PR DESCRIPTION
**Pull request by Piinks**
## Description

Continuing audit of skipped tests in the codebase.

This PR removes 35 skips, and closes out 7 TODOs.
There are 16 outstanding skips that are now notated with an appropriate issue.

## Related Issues

#32233
#32238
#41389
#42086
#44572
#56307
#56308
#56312
#56314
#56317

## Tests

Enabled!

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected one or more untriaged digests on patchset 1.
Please triage them at https://flutter-gold.skia.org/search?issue=55793.

**Comment by skia-gold** [comment]
Gold has detected one or more untriaged digests on patchset 2.
Please triage them at https://flutter-gold.skia.org/search?issue=55793.

**Comment by skia-gold** [comment]
Gold has detected one or more untriaged digests on patchset 3.
Please triage them at https://flutter-gold.skia.org/search?issue=55793.

**Comment by Piinks** [comment]
Closing temporarily as the gold endpoint is getting a lot of traffic from this draft PR.

File: packages/flutter/test/painting/gradient_test.dart [hunk 1]
```diff
@@ -817,15 +817,15 @@ void main() {
       for (final Gradient gradient in gradients45) {
         testWidgets('$gradient', (WidgetTester tester) async {
           await runTest(tester, gradient, 45);
-        }, skip: isBrowser); // TODO(yjbanov): web does not support golden tests yet: https://github.com/flutter/flutter/issues/40297
+        }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/gradient_test.dart:820]:
```suggestion
        }, skip: isBrowser); // https://github.com/flutter/flutter/issues/41389
```

File: packages/flutter/test/painting/gradient_test.dart [hunk 2]
```diff
@@ -817,15 +817,15 @@ void main() {
...
       }
     });
 
     group('Gradients - 90 degrees', () {
       for (final Gradient gradient in gradients90) {
         testWidgets('$gradient', (WidgetTester tester) async {
           await runTest(tester, gradient, 90);
-        }, skip: isBrowser); // TODO(yjbanov): web does not support golden tests yet: https://github.com/flutter/flutter/issues/40297
+        }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/gradient_test.dart:828]:
```suggestion
        }, skip: isBrowser); // https://github.com/flutter/flutter/issues/41389
```

File: packages/flutter/test/painting/notched_shapes_test.dart [hunk 3]
```diff
@@ -81,7 +81,7 @@ void main() {
           sampleSize: 100,
         ),
       );
-    }, skip: isBrowser);
+    }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/notched_shapes_test.dart:84]:
```suggestion
    }, skip: isBrowser); // https://github.com/flutter/flutter/issues/44572
```

File: packages/flutter/test/painting/text_painter_rtl_test.dart [hunk 4]
```diff
@@ -322,9 +322,7 @@ void main() {
         TextBox.fromLTRBD(0.0, 10.0, 10.0, 20.0, TextDirection.rtl), // Alef
       ],
     );
-  },
-  // Ahem-based tests don't yet quite work on Windows or some MacOS environments
-  skip: !isLinux);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/text_painter_rtl_test.dart:325]:
```suggestion
  }, skip: isBrowser); // https://github.com/flutter/flutter/issues/32238
```

**Comment by skia-gold** [comment]
Gold has detected one or more untriaged digests on patchset 4.
Please triage them at https://flutter-gold.skia.org/search?issue=55793.

**Comment by Piinks** [comment]
Images have been triaged, all that remains to complete here is applying issues to the remaining skips

File: packages/flutter/test/painting/text_style_test.dart [hunk 5]
```diff
@@ -238,7 +238,7 @@ void main() {
     expect(s2.apply().fontFamilyFallback, const <String>['Roboto', 'test']);
     expect(s2.apply(fontFamily: 'bar').fontFamily, 'bar');
     expect(s2.apply(fontFamilyFallback: const <String>['Banana']).fontFamilyFallback, const <String>['Banana']);
-  }, skip: isBrowser);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/text_style_test.dart:241]:
```suggestion
  }); // Reference needed
```

File: packages/flutter/test/painting/text_style_test.dart [hunk 6]
```diff
@@ -208,7 +208,7 @@ void main() {
 
     const TextStyle s10 = TextStyle(fontFamilyFallback: <String>[], package: 'p');
     expect(s10.fontFamilyFallback, <String>[]);
-  }, skip: isBrowser);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/text_style_test.dart:211]:
```suggestion
  }); // Reference needed
```

File: packages/flutter/test/painting/text_style_test.dart [hunk 7]
```diff
@@ -178,7 +178,7 @@ void main() {
     expect(ps2, equals(ui.ParagraphStyle(textAlign: TextAlign.center, fontWeight: FontWeight.w800, fontSize: 10.0, height: 100.0)));
     final ui.ParagraphStyle ps5 = s5.getParagraphStyle();
     expect(ps5, equals(ui.ParagraphStyle(fontWeight: FontWeight.w700, fontSize: 12.0, height: 123.0)));
-  }, skip: isBrowser);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/text_style_test.dart:181]:
```suggestion
  }); // Reference needed
```

File: packages/flutter/test/painting/text_painter_test.dart [hunk 8]
```diff
@@ -832,5 +831,5 @@ void main() {
       ui.Rect.zero,
     );
     expect(caretHeight, 50.0);
-  }, skip: kIsWeb);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/text_painter_test.dart:834]:
```suggestion
  }); // Reference needed
```

File: packages/flutter/test/painting/text_painter_test.dart [hunk 9]
```diff
@@ -729,7 +728,7 @@ void main() {
     expect(painter.inlinePlaceholderBoxes[11], const TextBox.fromLTRBD(250, 30, 300, 60, TextDirection.ltr));
     expect(painter.inlinePlaceholderBoxes[12], const TextBox.fromLTRBD(300, 30, 351, 60, TextDirection.ltr));
     expect(painter.inlinePlaceholderBoxes[13], const TextBox.fromLTRBD(351, 30, 401, 60, TextDirection.ltr));
-  }, skip: isBrowser);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/text_painter_test.dart:731]:
```suggestion
  }); // Reference needed
```

File: packages/flutter/test/painting/text_painter_test.dart [hunk 10]
```diff
@@ -143,7 +142,7 @@ void main() {
     expect(caretOffset.dx, 35);
     caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 2), ui.Rect.zero);
     expect(caretOffset.dx, 49);
-  }, skip: isBrowser);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/text_painter_test.dart:145]:
```suggestion
  }); // Reference needed
```

File: packages/flutter/test/painting/text_painter_test.dart [hunk 11]
```diff
@@ -121,7 +120,7 @@ void main() {
     expect(caretOffset.dx, 112); // 🇸
     caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 22), ui.Rect.zero);
     expect(caretOffset.dx, 112); // 🇸
-  }, skip: isBrowser);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/text_painter_test.dart:123]:
```suggestion
  }); // Reference needed
```

File: packages/flutter/test/painting/image_provider_test.dart [hunk 12]
```diff
@@ -443,7 +443,7 @@ void main() {
     const ImageConfiguration resizeConfig = ImageConfiguration(size: resizeDims);
     final Size resizedImageSize = await _resolveAndGetSize(resizedImage, configuration: resizeConfig);
     expect(resizedImageSize, resizeDims);
-  }, skip: isBrowser);
+  }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/image_provider_test.dart:446]:
```suggestion
  }); // Reference needed
```

File: packages/flutter/test/painting/image_provider_test.dart [hunk 13]
```diff
@@ -395,7 +395,7 @@ void main() {
           expect(events[i].cumulativeBytesLoaded, math.min((i + 1) * chunkSize, kTransparentImage.length));
           expect(events[i].expectedTotalBytes, kTransparentImage.length);
         }
-      }, skip: isBrowser);
+      }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/image_provider_test.dart:398]:
```suggestion
      }); // Reference needed
```

File: packages/flutter/test/painting/image_provider_test.dart [hunk 14]
```diff
@@ -65,7 +65,7 @@ void main() {
 
         expect(imageCache.statusForKey(provider).untracked, true);
         expect(imageCache.pendingImageCount, 0);
-      }, skip: isBrowser);
+      }, skip: isBrowser); // Reference needed
```
Comment (Piinks) [inline packages/flutter/test/painting/image_provider_test.dart:68]:
```suggestion
      }); // Reference needed
```

File: packages/flutter/test/painting/gradient_test.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/test/painting/gradient_test.dart:820]:
```suggestion
        }, skip: isBrowser); // https://github.com/flutter/flutter/issues/41389
```

File: packages/flutter/test/painting/gradient_test.dart (hunk 2 above)
Comment (Piinks) [reply packages/flutter/test/painting/gradient_test.dart:828]:
```suggestion
        }, skip: isBrowser); // https://github.com/flutter/flutter/issues/41389
```

File: packages/flutter/test/painting/notched_shapes_test.dart (hunk 3 above)
Comment (Piinks) [reply packages/flutter/test/painting/notched_shapes_test.dart:84]:
```suggestion
    }, skip: isBrowser); // https://github.com/flutter/flutter/issues/44572
```

File: packages/flutter/test/painting/text_painter_rtl_test.dart (hunk 4 above)
Comment (Piinks) [reply packages/flutter/test/painting/text_painter_rtl_test.dart:325]:
```suggestion
  }, skip: isBrowser); // https://github.com/flutter/flutter/issues/32238
```

**Review by goderbauer — APPROVED** [review]
LGTM 🎉

**Comment by ferhatb** [comment]
Thank you!!!

**Pull request merged**